lib/queue.js

```diff
@@ -0,0 +1,118 @@
+'use strict'
+
+var async = require('async')
+var chalk = require('chalk')
+var debug = require('debug-log')('mashape-analytics-queue')
+var http = require('http')
+var pkg = require('../package.json')
+var util = require('util')
+
+/**
+ * ALF Queue
+ */
+var Queue = module.exports = function (serviceToken, environment, options) {
+  var self = this
+
+  this.opts = options
+  this.timeout = null
+  this.userAgent = util.format('%s/%s', 'mashape-analytics-agent-node', pkg.version)
+
+  // setup event queue
+  // TODO specify worker pool
+  // TODO use msgpack + gzip?
+  this.queue = async.queue(function (entry, done) {
+    // append entry to log
+    self.alf.har.log.entries.push(entry)
+
+    // start timer
+    if (!self.timeout) {
+      self.timeout = setTimeout(function () {
```

Comment:
I'm confused. When else did you want the timer to start than after the first entry? Or did you want a heartbear timer; to flush every interval no matter what's in the bucket?

The currently logic is minimalistic and simple. There's no wasted cycles.

The first request starts the timer. Subsequent requests fill the bucket. If the timer expires, it flushes whatever is in the bucket and everything starts over.

If we start the timer before we add the entry, it could flush empty.

If there's a long period between requests, no cycles are lost in between. We're going for asynchronous signals. No ticks except what's necessary.

I don't see the benefit of separating any further than this. I'm concerned it may result in too much division, much that do much of nothing, but very nice architecture. A very Java-esque paradigm. You could just ignore this PR and do your own if you like.